src/connector/src/base.rs

```diff
@@ -167,97 +166,20 @@ pub trait SplitMetaData: Sized {
     fn restore_from_bytes(bytes: &[u8]) -> Result<Self>;
 }
 
-/// The persistent state of the connector.
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct ConnectorState {
-    pub identifier: Bytes,
-    pub start_offset: String,
-    pub end_offset: String,
-}
+pub type ConnectorState = Option<Vec<SplitImpl>>;
```

Comment:
Please add a comment here noting why it could be a None.

```suggestion
/// The persistent state of the connector. ....
pub type ConnectorState = Option<Vec<SplitImpl>>;
```